crates/floresta-chain/src/pruned_utreexo/consensus.rs

```diff
@@ -174,6 +190,99 @@ impl Consensus {
         }
         Ok(())
     }
+    /// Consumes the UTXOs from the hashmap, and returns the value of the consumed UTXOs.
+    /// If we do not find the UTXO, we return an error invalidating the input that tried to 
+    /// consume that UTXO.
+    fn consume_utxos(input: &TxIn ,utxos: & mut HashMap<OutPoint, TxOut>, value_var: &mut u64) -> Result<(), BlockchainError> {
+        match utxos.get(&input.previous_output) {
+            Some(prevout) => {
+                *value_var += prevout.value.to_sat();
+                utxos.remove(&input.previous_output);
+            }
+            None => {
+                return Err(BlockValidationErrors::InvalidTx(alloc::format!(
+                    //This is the case when the spender:
+                    // - Spends an UTXO that doesn't exist
+                    // - Spends an UTXO that was already spent
+                    "Invalid input: {:?}",
+                    input.previous_output
+                ))
+                .into());
+            }
+        };
+        Ok(())
```

Comment:
I do not have any idea of how to test it.
@lla-dane Do you have any idea ? Since you are working on some tests.

Reply:
@Davidson-Souza as you can see, to avoid double spending, the `consumes_utxos()` actually delete the utxo from the hashmap. Hope this doesnt brake anything outside this function... 
If it does, i see a solution where, in `consumes_utxos()` we could accumulate the utxos that been spent and after validating the block applying the difference to it.

Reply:
I could be done like this: You simulate an environment where you connect, lets say 2-3 blocks keeping it short, and you get the UTXO HashMap from the chain state, or wherever it is being stored and while connecting the next block, you see if this function return `Ok()`, I think this is one way of doing it.

Reply:
Or simply if you can get the UTXO hashmap directly, then just pass those in the function, and assert the result for the immendiatly next transaction

Reply:
You are gonna need raw transactions from a specific network for which you want to test it, most of the places everything is tested for Signet. You can get that raw data from `utreexod`.

Reply:
> You can get that raw data from utreexod.

Yes, i was more worried about that arbitrary data. I think that collecting it from Bitcoin Core is good doing too.
maybe with `BitcoinFuzz` we can cover more specific cases 

>Or simply if you can get the UTXO hashmap directly, then just pass those in the function, and assert the result for the immendiatly next transaction

I think thats the best approach since i just want to test the validation code.

Reply:
> Yes, i was more worried about that arbitrary data. I think that collecting it from Bitcoin Core is good doing too.
maybe with BitcoinFuzz we can cover more specific cases

I think you should you `utreexod` node, for getting data since we are also dealing with Utreexo accumulator here, and also FLoresta's node accept UtreexoBlocks from its peers and UtreexoState for that particular block also, and then updates the accumulator to keep up with the UTXOs.

Reply:
This would be more valuable while doing integration tests while getting test-data directly from a utreexo acc.
But I'll make sure to use them while building the unitary battery

Reply:
> Yes, i was more worried about that arbitrary data.

I don't think this is a problem. You have total control over the utxos passed to this function, take a random block, duplicate one transaction (double-spend) for one test, remove one utxo for another...

> This would be more valuable while doing integration tests while getting test-data directly from a utreexo acc.

No, the inputs are only used by `verify_block_transactions`. Just leave the script validation after doing things like adding-up fees.

Reply:
>I don't think this is a problem. You have total control over the utxos passed to this function, take a random block, duplicate one transaction (double-spend) for one test, remove one utxo for another...

Yeah, but the design of the code will not help in this case. Itll only fail, we will not know why and what triggered that. Not enough info, not good.

Reply:
You can change it to return an `enum` with the actual error